### PR TITLE
refactor tests

### PR DESCRIPTION
**Pull request by lauherkcambria**
#### What tests were added (if relevant), and do all tests (rspec, karma, and integration CALS tests) pass?
- a bunch of rspec testing was added in order to bring rspec code coverage to 98 from 72, fix vcr set up and some clean up from previous branches

note: residence spec and applicant spec will need to be gone over once we have a better clear strategy on how to handle this. 

#### Which User story does this relate to?
- this is related to technical debt incurred from previous user stories.

#### Brief feature description
- bring test coverage up

#### Does this PR block/effect other stories?
- no

#### Do any commands needs to be run in order to run the project with this PR?
- rake tmp:cache:clear rails assets:precompile

#### Code coverage percent diff as a result of this PR?
- 72 -> 98

#### JS errors diff as a result of this PR?
- na

### CODE REVIEW

**Comment by lauherkcambria** [comment]
jenkins retest this please.

**Comment by ratneshraval** [comment]
I know why this is failing. Let me adjust the config.

File: config/routes.rb [hunk 1]
```diff
@@ -13,26 +13,15 @@
 
 
   namespace :rfa do
-    resources :a01, only: [ :show, :create, :edit, :update] do
+    resources :a01 do
 
       resources :applicant, only: [:index, :create, :edit]
-      resource :residence, only: [:show, :create, :edit]
+      resources :residence, only: [:show, :create, :edit]
```
Comment (ratneshraval) [inline config/routes.rb]: This should stay as singular resource since there is only 1 residence per application.

File: spec/support/vcr_setup.rb [hunk 2]
```diff
@@ -14,7 +14,7 @@
 private
 def localhost_has_identify?(request)
   #byebug
-  if request.uri.starts_with?(CALS_API_BASE_URL)
+  if request.uri.starts_with?(CALS_API_URL)
```
Comment (ratneshraval) [inline spec/support/vcr_setup.rb]: This needs to look at both CALS_API_BASE_URL and CALS_API_URL for cassettes to work.

**Comment by lauherkcambria** [comment]
After this PR is merged, lets make it a priority to continue with keeping up on adding tests when creating new features so that code coverage doesn't drop under 90 percent please.

File: spec/support/vcr_setup.rb [hunk 3]
```diff
@@ -7,23 +7,6 @@
   c.allow_http_connections_when_no_cassette = true
   c.ignore_localhost = false
   c.ignore_request do |request|
-    localhost_has_identify?(request)
+  !(request.uri.starts_with?(CALS_API_BASE_URL) || request.uri.starts_with?(CALS_API_URL))
```
Comment (ratneshraval) [inline spec/support/vcr_setup.rb]: :)

Comment (lauherkcambria) [inline spec/support/vcr_setup.rb]: ;)

**Pull request merged**